Project/UISS/UIView+UISS.m

```diff
@@ -22,4 +25,102 @@ - (void)setBackgroundUISSColor:(UIColor *)backgroundColor;
     [self setBackgroundColor:backgroundColor];
 }
 
+static IMP ObjectMethodSetImplementation(id obj, SEL sel, id implemetationBlock)
+{
+    Class class = object_getClass(obj);
+
+    Method method = class_getInstanceMethod(class, sel);
+
+    NSCParameterAssert(method != nil);
+
+    const char *methodEncoding = method_getTypeEncoding(method);
+
+    IMP implementation = imp_implementationWithBlock(implemetationBlock);
+
+    if (!class_addMethod(class, sel, implementation, methodEncoding)) {
+        return method_setImplementation(method, implementation);
+    }
+
+    return NULL;
+}
+
+static UISSAppearanceProxy *AppearanceProxy(Class class, NSString *identifier)
+{
+    if (!identifier) {
+        return nil;
+    }
+
+    static void *AssociationKey = &AssociationKey;
+    NSMutableDictionary<NSString *, UISSAppearanceProxy *> *proxies = objc_getAssociatedObject(class, AssociationKey);
+
+    if (!proxies) {
+        proxies = [NSMutableDictionary new];
+        objc_setAssociatedObject(class, AssociationKey, proxies, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+    }
+
+    __auto_type proxy = proxies[identifier];
+
+    if (!proxy) {
+        proxy = [UISSAppearanceProxy proxyForObjectsOfClass:class];
+        proxies[identifier] = proxy;
+    }
+
+    return proxy;
+}
+
+static void *UISSIdentifierKey = &UISSIdentifierKey;
+
+- (NSString *)UISSIdentifier
+{
+    return objc_getAssociatedObject(self, UISSIdentifierKey);
+}
+
+- (void)setUISSIdentifier:(NSString *)identifier
+{
+    objc_setAssociatedObject(self, UISSIdentifierKey, identifier, OBJC_ASSOCIATION_COPY_NONATOMIC);
+
+    static void *TweakedKey = &TweakedKey;
+
+    Class objClass = [self class];
+
+    if (objc_getAssociatedObject(objClass, TweakedKey)) {
+        return;
+    }
+
+    SEL selector = @selector(didMoveToWindow);
+
+    __block void (*originalDidMoveToWindow)(__unsafe_unretained id, SEL) = NULL;
+
+    originalDidMoveToWindow = ObjectMethodSetImplementation(self, selector, ^(__unsafe_unretained UIView *self) {
```

Comment:
What happens if the setter is invoked multiple times?

Reply:
Swizzling is happening only once per class, however it doesn't check superclass. I think it makes sense to tweak UIView class directly.